### PR TITLE
Update README about directory structure and validation errors

### PR DESCRIPTION
**Pull request by jeremylinlin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,13 +30,32 @@ $ operator-courier push $BUNDLE_DIR $EXAMPLE_NAMESPACE $EXAMPLE_REPOSITORY $EXAM
 
 Once that is created, you should be able to view your bundle on quay.io's Application page for your particular namespace, repo, and release version (https://quay.io/application/$EXAMPLE_NAMESPACE/$EXAMPLE_REPOSITORY?tab=$EXAMPLE_RELEASE)
 
-For more info, run help on any of the subcommands
+For more info, run help on the main program or any of the subcommands
 
 ```bash
 $ operator-courier -h
-$ operator-courier push -h
+$ operator-courier SUBCOMMAND -h
```
Comment (kevinrizza) [inline README.md]: Nit: `$SUBCOMMAND`

File: README.md [hunk 2]
```diff
@@ -30,13 +30,32 @@ $ operator-courier push $BUNDLE_DIR $EXAMPLE_NAMESPACE $EXAMPLE_REPOSITORY $EXAM
...
 ```
 
+### Source Directory Structure
+
+Currently operator-courier supports source directory in both **flat** and **nested** structure.
+
+- **flat directory** includes CRD, CSV and package files all in one single directory
```
Comment (kevinrizza) [inline README.md]: IMO we shouldn't advertise the flat dir structure. We have it for legacy purposes but all of our docs should be pointing to the nested operator-registry format.

Comment (jeremylinlin) [reply README.md]: @kevinrizza But in the current README in the [CLI](https://github.com/operator-framework/operator-courier/blob/master/README.md#command-line-interface) and [Library](https://github.com/operator-framework/operator-courier/blob/master/README.md#library) section, they only assume the source directory to contain our old flat directory format. Should I also update those and replace them with the new nested format?

Comment (kevinrizza) [reply README.md]: @jeremylinlin oh yeah that's a good point. We should update those places in the readme as well.

**Comment by jeremylinlin** [comment]
@kevinrizza @SamiSousa Updated the PR. Please take a look when you have a chance. Thanks!

File: README.md [hunk 3]
```diff
@@ -16,27 +16,40 @@ To install the operator courier project to use from the command line, just insta
 $ pip3 install operator-courier
 ```
 
-Once the project is installed, you can verify that a set of files is valid and can be bundled and pushed. First, create a flat directory containing all of the CSV, CRD and PACKAGE files that are included in your bundle.  Then just use `operator-courier verify` to test it.
+Once the project is installed, you can run `verify` command on a set of operator manifests with different versions in the source directory. The source directory includes a package file in the source directory root, and version folders containing CRD and CSV files required for that manifest version (See [Manifest format](https://github.com/operator-framework/operator-registry#manifest-format) for more details)
```
Comment (awgreene) [inline README.md]: This is a very convoluted sentence. I recommend rewriting it to just reference the manifest format:
>Once the project is installed, you can run **the** `verify` command on a directory that adheres to the expected [Manifest format](https://github.com/operator-framework/operator-registry#manifest-format).

File: README.md [hunk 4]
```diff
@@ -16,27 +16,40 @@ To install the operator courier project to use from the command line, just insta
...
 
 Once that is created, you should be able to view your bundle on quay.io's Application page for your particular namespace, repo, and release version (https://quay.io/application/$EXAMPLE_NAMESPACE/$EXAMPLE_REPOSITORY?tab=$EXAMPLE_RELEASE)
 
-For more info, run help on any of the subcommands
+For more info, run help on the main program or any of the subcommands
 
 ```bash
 $ operator-courier -h
-$ operator-courier push -h
+$ operator-courier $SUBCOMMAND -h
```
Comment (awgreene) [inline README.md]: Do we need this?

File: README.md [hunk 5]
```diff
@@ -16,27 +16,40 @@ To install the operator courier project to use from the command line, just insta
...
 
-To generate an operator bundle and push it to a quay.io app registry just use `operator-courier push`. Just pass the directory, namespace, repository, release version and quay.io authorization token needed to push.
+To push the operator manifests to a quay.io app registry just use `operator-courier push`. Just pass the directory, namespace, repository, release version and quay.io authorization token needed to push.
 
 ```bash
-$ operator-courier push $BUNDLE_DIR $EXAMPLE_NAMESPACE $EXAMPLE_REPOSITORY $EXAMPLE_RELEASE "$AUTH_TOKEN"
+$ operator-courier push $MANIFESTS_DIR $EXAMPLE_NAMESPACE $EXAMPLE_REPOSITORY $EXAMPLE_RELEASE "$AUTH_TOKEN"
 ```
 
 Once that is created, you should be able to view your bundle on quay.io's Application page for your particular namespace, repo, and release version (https://quay.io/application/$EXAMPLE_NAMESPACE/$EXAMPLE_REPOSITORY?tab=$EXAMPLE_RELEASE)
```
Comment (awgreene) [inline README.md]: `... **bundle** on quay.io's Application ...`

Should we use the term `bundle` here?

File: README.md (hunk 4 above)
Comment (jeremylinlin) [reply README.md]: @awgreene Do you think I should remove `$ operator-courier $SUBCOMMAND -h`? But the help message will be different in those two cases.

File: README.md (hunk 3 above)
Comment (jeremylinlin) [reply README.md]: @awgreene Thanks for the suggestion! Updated.

File: README.md (hunk 5 above)
Comment (jeremylinlin) [reply README.md]: Thanks @awgreene. Great catch!

File: README.md (hunk 4 above)
Comment (awgreene) [reply README.md]: I believe that it's overkill given that we mention that the subcommands support the `-h`  flag directly before the example. It's a very small nit so don't feel like it has to be removed.

**Pull request merged**